File: 1/deriv.js
function fx(x){
    let fx = x ** 2 - 1.0;
    return fx;
}

function DIF_F(fx2, xd){
    let h = 10 ** -9;
    let dif_f = (fx2(xd + h) - fx2(xd))/ h;
    return dif_f;
}

let x = 1.0;
console.log(DIF_F(fx, x));

File: 1/deriv.c
#include <stdio.h>
#include <math.h>
double DIF_F(double (*fun)(double x),double xd);
double fx(double x);
int main(void){
    double x;
    x = 1.0;
    printf("df/dx=%lf\n",DIF_F(fx,x));
    return 0;
}

double fx(double x){
    double fx;
    fx = pow(x,2.0)-1.0;
    return fx;
}

double DIF_F(double (*fun)(double x),double xd){
    double DIF_F;
    double h = pow(10.0,-9.0);
    DIF_F = (fun(xd+h)-fun(xd))/h;
    return DIF_F;
}